### PR TITLE
23 typeahead

### PR DESCRIPTION
**Pull request by austingardner**
SQUASH please!

This creates a reusable typeahead function, that will be added into the menu component in ebayui-core.

I want to merge this first so we can add the updated makeup version into ebayui-core before I put up a PR for the updated menu component. 

I already demo'd this feature so I don't think I need to add gifs or screenshots but I can if needed. 

As far as the coveralls check goes -> it seems my build passed but then merging the new files dropped the coverage. So maybe it's okay? If not I can add some more tests.

### CODE REVIEW

**Review by ianmcburnie — CHANGES_REQUESTED** [review]
Could you please add a `README.md` file for the new module. It should give a brief description of its purpose & how to use it. Check the other modules for examples of formatting.

File: docs/makeup-typeahead/index.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        <title>makeup-typeahead demo</title>
+        <meta name="viewport" content="width=device-width, initial-scale=1"/>
+    </head>
+    <body>
+        <main>
+            <h1>makeup-typeahead demo</h1>
+            <span>just start typing to see the typeahead's results!</span>
+            <h2>Country List</h2>
+            <ul>
+                <li>Albania</li>
```
Comment (ianmcburnie) [inline docs/makeup-typeahead/index.html:13]: Nitpick, but could you please put these in alphabetical order :-)

**Comment by ianmcburnie** [comment]
Seeing a JS error in IE11 (yes, sorry we have to support IE11 still). You can either use an older API method or include a polyfill (e.g core-js).

Screen Shot 2020-11-18 at 6 25 06 PM

**Comment by austingardner** [comment]
The makeup project is small enough.. could we just add babel really quickly? We could use a CDN script from something like unpkg to apply babel before we export it

**Comment by ianmcburnie** [comment]
Actually, the project is already using babel. So a few options I think:

1. Add a babel plugin to devDependencies (if you can find one)
1. Require and use the relevant corejs shim
1. Require and use a polyfill (if you can find one)

**Comment by austingardner** [comment]
Added a shim from core-js. Can you double check it works on IE 11 now?

**Comment by ianmcburnie** [comment]
I tested it and the shims are working, thanks.

File: packages/makeup-typeahead/package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "makeup-typeahead",
+  "description": "Captures letter keys pressed and returns a list of relevant strings",
+  "version": "0.1.0",
```
Comment (ianmcburnie) [inline packages/makeup-typeahead/package.json:4]: Nitpick, but can you drop this to 0.0.1? After that I think we are good to publish to NPM!

Comment (austingardner) [reply packages/makeup-typeahead/package.json]: done!

**Review by ianmcburnie — APPROVED** [review]
Looks great! I'll merge and do the initial publish to NPM so that it appears under my account, but in future I'll probably have to shuffle all of these modules under a package namespace (for better team/account management).

**Pull request merged**